src/containers/Tenant/Diagnostics/TopQueries/TopQueries.tsx

```diff
@@ -93,10 +136,30 @@ export const TopQueries = ({tenantName}: TopQueriesProps) => {
         dispatch(setTopQueriesFilters({text}));
     };
 
+    const handleTimeFrameChange = (value: string[]) => {
+        setTimeFrame(value[0] as TimeFrame);
+    };
+
     const handleDateRangeChange = (value: DateRangeValues) => {
         dispatch(setTopQueriesFilters(value));
     };
 
+    // Get columns based on query mode
+    const columns: Column<any>[] = React.useMemo(() => {
+        return isTopQueries ? getTopQueriesColumns() : getRunningQueriesColumns();
+    }, [isTopQueries]);
+
+    // Use selected columns hook
+    const {columnsToShow, columnsToSelect, setColumns} = useSelectedColumns(
+        columns,
+        isTopQueries
```

Comment:
Instead of ternary operator lets try to do two separate components here

Reply:
You can call this hook twice, no need in ternaries
```
const {columnsToShow: topQueriesColumnsToShow. ...rest} = useSelectedColumns()
const {columnsToShow: runningQueriesColumnsToShow. ...rest} = useSelectedColumns()
```